src/analysis/retail/mage/fire/guide/Combustion.tsx

```diff
@@ -253,6 +257,24 @@ class CombustionGuide extends Analyzer {
               castEntries={this.combustionData}
             />
           </div>
+          <div>
+            <DamageContribution
+              title="Damage During Combustion"
+              spells={[
+                { spell: TALENTS.PYROBLAST_TALENT, color: '#ff6600' },
+                { spell: SPELLS.FIRE_BLAST, color: '#ff9933' },
+                { spell: SPELLS.PHOENIX_FLAMES_DAMAGE, color: '#ffcc00' },
+                { spell: SPELLS.IGNITE, color: '#910808ff' },
+                { spell: SPELLS.FLAMESTRIKE, color: '#cc3300' },
+                { spell: SPELLS.ARCANE_PHOENIX_DAMAGE, color: '#7b1d92ff' },
+              ]}
+              calculateContribution={(spellId: number) =>
+                this.combustionDamageTracker.getDamageForSpell(spellId)
+              }
```

Comment:
so as an example of the issue with `calculateContribution`: this could be replaced by:

1. calling `[ ... ].map(entry => ({ ...entry, value: getDamageForSpell(entry.spell) }))`
2. passing `totalValue={combustionDamageTracker.totalDamage}`

with virtually no additional work on the consumer side, but without the added logic & complexity in the `DamageContribution` component.